pmlc/dialect/pxa/transforms/stencil_gemm.cc

```diff
@@ -58,7 +60,9 @@ struct GemmOperand {
 class StencilGEMM : public StencilBase {
 private:
   unsigned numThreads;
+  std::string strategy;
```

Comment:
Let's use an `enum` instead of a `string`. Generating an enum with tblgen will give us the ability to convert to/from a string.

Reply:
Hi @flaub : Did you mean to declare the strategy like it's done for "pxa-vectorize": https://github.com/plaidml/plaidml/blob/brgemm_polydl/pmlc/dialect/pxa/transforms/passes.td#L129 ? 

But, "x86-affine-stencil-xsmm" pass is not declared in pxa/transforms/passes.td. Instead it's declared in target/x86/passes.td. From target/x86/pipeline.cc, pxa/transforms/stencil_gemm.cc is invoked. I had tried adding options to pmlc/target/x86/passes.td here: https://github.com/plaidml/plaidml/blob/brgemm_polydl/pmlc/target/x86/passes.td#L25. Curiously, the options were not being read. I am not sure why.

Reply:
No, I mean we should create an enum with tblgen and then use it to define the available strategies. The pass would still be configured with a string, it's just that the string can be converted into the enum and the rest of the code would use the enum.

Reply:
Why do we need a strategy in the first place? It seems like we should have a cost model that decided when to use a batched vs non-batched GEMM. I guess the question is, why would we ever want to use a non-batched GEMM? Is there some overhead/cost to doing so?

Reply:
Yeah, in the future it would be nice to have a cost function which will use one or the other variant automatically. For now, I think that there is value to having both of them and switching between them manually. Which one will perform better would be a function of cache sizes and parallelization. Additionally, the quality of microkernels JITed, if there is a difference, would determine which one would be better.

Reply:
@flaub : Hope the `enum` thing is not a blocker. For my future reference, do you have an example you could point me to? Thanks